data-jdbc/src/main/java/io/micronaut/data/jdbc/operations/DefaultJdbcRepositoryOperations.java

```diff
@@ -158,6 +163,7 @@ public final class DefaultJdbcRepositoryOperations extends AbstractSqlRepository
 
     private final ColumnNameCallableResultReader columnNameCallableResultReader;
     private final ColumnIndexCallableResultReader columnIndexCallableResultReader;
+    private final Map<Dialect, SqlExceptionMapper> sqlExceptionMappers = new HashMap<>(Dialect.values().length);
```

Comment:
Use EnumMap, we should probably support multiple remappers of the same dialect

Reply:
Changed to EnumMap and allowing list of mappers. Mappers extend Ordered if there would be some with higher priorities to be resolved first.